File: 2075-decode-the-slanted-ciphertext/2075-decode-the-slanted-ciphertext.cpp
class Solution {
public:
    string decodeCiphertext(string encodedText, int rows) {
        string res = "";
        int cols = encodedText.length() / rows;
        for(int idx = 0; idx < cols; idx++)
        {
            int i = 0;
            int j = idx;
            while(i < rows && j < cols)
            {
                res += encodedText[i * cols + j];
                i++;
                j++;
            }
        }
        while(!res.empty() && res.back() == ' ')
        {
            res.pop_back();
        }
        return res;
    }
};

File: 2075-decode-the-slanted-ciphertext/2075-decode-the-slanted-ciphertext.py
class Solution:
    def decodeCiphertext(self, encodedText: str, rows: int) -> str:
        res = ""
        cols = len(encodedText) // rows
        if cols == 0:
            return ""
        textMat = [list(encodedText[i * cols : i * cols + cols]) for i in range(rows)]
        for idx in range(cols):
            i = 0
            j = idx
            while i < rows and j < cols:
                res += textMat[i][j]
                i += 1
                j += 1
        return res.rstrip()